### DISCUSSION TITLE
Question about PBF water particles.

### DISCUSSION THREAD

**Discussion by DaveGravel** [NVIDIA-Omniverse/PhysX#50] [Q&A] [answered]
Hi,
I made a water simulation demo based on the SnippetPBF demo from the sdk.
The demo works pretty well.
I added dynamic bodies in the simulation to interact with water.
It seems to work well too.
I have a problem with some dynamic bodies.
If I start moving bodies in water, Some bodies are affected by the one I'm moving.
It seems to affect the body with higher mass.
I don't know if it's a bug or a misconfiguration.
Does anyone have any idea what could be causing this behavior ?
Here a video for show the problem.
https://www.youtube.com/watch?v=TvClDGX96Jk
Thanks.

**Comment by amoravanszky** [comment, marked as answer]
Its an interesting effect.  My first idea is that you are creating such big shocks in the liquid by your super rapid movements of the masses that it is creating massive shockwaves of particle momentum in the liquid that the simulation is very much subsampling because you are using a relatively coarse simulation step.  At least the energy that you are putting into the fluid by moving one object doesn't seem totally disproportionate to the energy required to shoot the other object upward.  I would be curious if the problem still reproduces to the same extent if you reduce the time step size by a factor of 10.  Of course this might lead to you no longer being able to simulate in real time ... but there is no free lunch.

Of course it is also totally possible that it is a bug.

**Reply by DaveGravel** [reply]
Thanks, 
Ok I have make some changes for test.
`My first idea is that you are creating such big shocks in the liquid by your super rapid movements of the masses`
Yes, I have think the same thing me too at first.
`if you reduce the time step size by a factor of 10. Of`
I use a loop of 1/60 I have try 1/30 and 1/120 and I get pretty much the same result.
Or maybe you talk about a other thing.

The particle system Solver Iteration is set to default, iter: 4 vel: 1
I have try eTGS and ePGS solver it is pretty much the same result too.
If I set the fluid density higher the effect look more present.
I have now set all dynamics objects in the water to 15 of mass.
I think you can see more the problem in this video.
In this video I use a loop of 1/60
https://www.youtube.com/watch?v=24RZM0ecgDA
In the video, it looks like it only hits objects above water. But it can affect all objects in the water. 
The problem seems more present when I move the object to certain places in the water.
Sorry about my english, I'm not really good for explain things.
